lib/src/core/guild/auto_moderation.dart

```diff
@@ -0,0 +1,305 @@
+  /// The type of the action.
+  ActionTypes get actionType;
+
+  /// Additionnal metadata needed during execution for this specific action type.
+  IActionMetadata? get actionMetadata;
+}
+
+abstract class IActionMetadata {
+  /// The channel if to wich user content should be logged.
+  /// The associated action type is [ActionTypes.sendAlertMessage].
+  Cacheable<Snowflake, ITextGuildChannel>? get channelId;
+
+  /// The timeout duration - maximum duration is 4 weeks (2,419,200 seconds).
+  /// It's associated action type is [ActionTypes.timeout].
+  Duration? get duration;
+}
+
+class AutoModerationRule extends SnowflakeEntity implements IAutoModerationRule {
+  @override
+  late final Cacheable<Snowflake, IGuild> guild;
+
+  @override
+  late final String name;
+
+  @override
+  late final Cacheable<Snowflake, IMember> creator;
+
+  @override
+  late final EventTypes eventType;
+
+  @override
+  late final TriggerTypes triggerType;
+
+  @override
+  late final ITriggerMetadata triggerMetadata;
+
+  @override
+  late final List<IActionStructure> actions;
+
+  @override
+  late final bool enabled;
+
+  @override
+  late final List<RoleCacheable> ignoredRoles;
+
+  @override
+  late final List<ChannelCacheable<ITextGuildChannel>> ignoredChannels;
+
+  AutoModerationRule(RawApiMap rawData, INyxx client) : super(Snowflake(rawData['id'])) {
+    guild = GuildCacheable(client, Snowflake(rawData['guild_id']));
+    name = rawData['name'] as String;
+    creator = MemberCacheable(client, Snowflake(rawData['creator_id']), guild);
+    eventType = EventTypes._fromValue(rawData['event_type'] as int);
+    triggerType = TriggerTypes.fromValue(rawData['trigger_type'] as int);
+    triggerMetadata = TriggerMetadata(rawData['trigger_metadata'] as RawApiMap);
+    actions = [...?(rawData['actions'] as RawApiList?)?.map((a) => ActionStructure(a as RawApiMap, client))];
+    enabled = rawData['enabled'] as bool;
+    ignoredRoles = (rawData['exempt_roles'] as RawApiList).isNotEmpty
+        ? [...(rawData['exempt_roles'] as RawApiList).map((r) => RoleCacheable(client, Snowflake(r), guild))]
+        : [];
+    ignoredChannels = (rawData['exempt_channels'] as RawApiList).isNotEmpty
+        ? [...(rawData['exempt_channels'] as RawApiList).map((r) => ChannelCacheable(client, Snowflake(r)))]
+        : [];
+  }
+
+  @override
+  String toString() => 'IAutoModerationRule(id: $id, guildId: ${guild.id}, name: $name, triggerMetadata: $triggerMetadata)';
+}
+
+class TriggerMetadata implements ITriggerMetadata {
+  // Maybe return null instead of empty list
+  @override
+  late final List<KeywordPresets>? keywordPresets;
+
+  @override
+  late final List<String>? keywordsFilter;
+
+  @override
+  late final List<String>? allowList;
+
+  @override
+  late final int? mentionLimit;
+
+  /// Creates an instance of [TriggerMetadata]
+  TriggerMetadata(RawApiMap data) {
+    keywordsFilter = data['keyword_filter'] != null ? [...data['keyword_filter']] : null;
```

Comment:
why its done that way? cant it just be `data['keyword_filter'] != null ? data['keyword_filter'] : null;`